From 4c4f5f2274bfe174e82d7c68627fb6c95b04fd8b Mon Sep 17 00:00:00 2001
From: Sandro <devel@penguinpee.nl>
Date: Wed, 15 Jan 2025 20:59:47 +0100
Subject: [PATCH 1005/1005] Revert "bottle-row: Port to AdwWrapBox"

AdwWrapBox is still deemed unstable and only available since 1.7.

This reverts commit eb81c0306ef20aa5e4c0104a0a6d55779996b027.
---
 bottles/frontend/ui/bottle-row.blp | 16 ++++++++++------
 bottles/frontend/views/list.py     |  6 +++---
 2 files changed, 13 insertions(+), 9 deletions(-)

diff --git a/bottles/frontend/ui/bottle-row.blp b/bottles/frontend/ui/bottle-row.blp
index 163900ea..2598ed52 100644
--- a/bottles/frontend/ui/bottle-row.blp
+++ b/bottles/frontend/ui/bottle-row.blp
@@ -5,13 +5,17 @@ template $BottlesBottleRow: Adw.ActionRow {
   activatable: true;
   use-markup: false;
 
-  Adw.WrapBox wrap_box {
-    valign: center;
+  Box {
+    spacing: 6;
 
-    styles [
-      "tag",
-      "caption",
-    ]
+    Label label_env {
+      valign: center;
+
+      styles [
+        "tag",
+        "caption",
+      ]
+    }
   }
 
   Button button_run {
diff --git a/bottles/frontend/views/list.py b/bottles/frontend/views/list.py
index a8f6c587..91a52e2b 100644
--- a/bottles/frontend/views/list.py
+++ b/bottles/frontend/views/list.py
@@ -37,7 +37,7 @@ class BottlesBottleRow(Adw.ActionRow):
 
     # region Widgets
     button_run = Gtk.Template.Child()
-    wrap_box = Gtk.Template.Child()
+    label_env = Gtk.Template.Child()
 
     # endregion
 
@@ -74,8 +74,8 @@ class BottlesBottleRow(Adw.ActionRow):
         self.set_title(self.config.Name)
         if self.window.settings.get_boolean("update-date"):
             self.set_subtitle(update_date)
-
-        self.wrap_box.append(Gtk.Label.new(self.config.Environment))
+        self.label_env.set_text(_(self.config.Environment))
+        self.label_env.add_css_class("tag-%s" % self.config.Environment.lower())
 
         # Set tooltip text
         self.button_run.set_tooltip_text(_(f"Run executable in “{self.config.Name}”"))
-- 
2.49.0

